server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1446,14 +1457,17 @@ public synchronized void updateFromClusterManager(
                             + " as in-sync but it does not exist locally";
                         final long localCheckpoint = SequenceNumbers.UNASSIGNED_SEQ_NO;
                         final long globalCheckpoint = localCheckpoint;
+                        final boolean assignedToRemoteStoreNode = indexSettings.isRemoteNode()
```

Comment:
lets add comments here.

Reply:
Changed this to `indexSettings.isRemoteStoreEnabled()` which was my intention in the first place. This is done so that for indexes with remote based index settings already present, we don't iterate through the DiscoveryNodes list